Update teamcity setup config docs to reflect basic auth provision (#15515)
DataDog/integrations-core: teamcity/README.md

#### Prepare TeamCity

You can enable [Guest login](#guest-login), or identify [user credentials](#user-credentials) for basic HTTP authentication.

##### Guest login

> 1. [Enable guest login][2] or identify user credentials (with required permissions) for basic HTTP authentication.

Reviewer: ```suggestion
1. [Enable guest login][2].
```

#### Prepare TeamCity

> 1. [Enable guest login][2] or identify user credentials (with required permissions) for basic HTTP authentication.

Reviewer: ```suggestion
You can enable [Guest login](#guest-login), or identify [user credentials](#user-credentials) for basic HTTP authentication.

##### Guest login

1. [Enable guest login][2] or identify user credentials (with required permissions) for basic HTTP authentication.
```

Author: Thank you @buraizu , accepted the change.

![Guest user settings][20]
![Assign Role][21]

**Note:**
- For basic HTTP authentication specify identified `username` and `password` in `teamcity.d/conf.yaml` at the `conf.d/` folder of your [Agent's configuration directory][3] 

> - `Access denied. Enable guest authentication or check user permissions.` would result in case of incorrect/insufficent user permission

Reviewer 2: Do you think we could also mention the permissions related to `Agent Workload Statistics` at the same time? https://github.com/DataDog/architecture/blob/master/rfcs/agent-integrations/teamcity/teamcity.md#authentication

Author: Updated with the feedback from docs team.

> **Note:**
> - For basic HTTP authentication specify identified `username` and `password` in `teamcity.d/conf.yaml` at the `conf.d/` folder of your [Agent's configuration directory][3] 
> - `Access denied. Enable guest authentication or check user permissions.` would result in case of incorrect/insufficent user permission

Reviewer: ```suggestion
##### User credentials

For basic HTTP authentication, specify an identified `username` and `password` in the `teamcity.d/conf.yaml` file in the `conf.d/` folder of your [Agent's configuration directory][3]. If you encounter an `Access denied. Enable guest authentication or check user permissions.` error, ensure the user has the correct permissions.
```

![Guest user settings][20]
![Assign Role][21]

##### User credentials

For basic HTTP authentication
- Specify an identified `username` and `password` in the `teamcity.d/conf.yaml` file in the `conf.d/` folder of your [Agent's configuration directory][3].

> - If you encounter an `Access denied. Enable guest authentication or check user permissions.` error, ensure the user has the correct permissions.

Reviewer: ```suggestion
- If you encounter an `Access denied. Enable guest authentication or check user permissions.` error, ensure the user has the correct permissions:
```
